pkg/client/clientset/versioned/typed/jenkins.io/v1/app_expansion.go

```diff
@@ -28,6 +28,9 @@ func (c *apps) PatchUpdate(app *v1.App) (*v1.App, error) {
 	if err != nil {
 		return nil, err
 	}
+	if patch == nil {
```

Comment:
Have you verified this? I don't think the patch will ever be nil. If the two versions are identical '[]' is returned. S
 something like `if string(patch) == "[]" ` would do. Could you verify this?

Other than that I am all up for saving some unnecessary calls.

Reply:
Well, I changed util.CreatePatch to return nil instead to make these if statements simpler. Or so I thought. I had missed the fact that if you return nil as an array it actually becomes an empty array and not nil. So I I'll revert that change in util.CreatePatch and change the if statements. I think it is uglier, but I can't think of a prettier solution right now.